src/1password.ts

```diff
@@ -0,0 +1,178 @@
+import * as core from '@actions/core'
+// import * as exec from '@actions/exec'
+import {install} from './install'
+import * as tc from '@actions/tool-cache'
+import {execWithOutput} from './exec'
+import {Item} from './types'
+import * as parsing from './parsing'
+
+const ONE_PASSWORD_VERSION = '1.8.0'
+
+export class OnePassword {
+  onePasswordEnv: {[key: string]: string}
+
+  constructor(deviceId: string) {
+    this.onePasswordEnv = {
+      OP_DEVICE: deviceId
+    }
+
+    if (process.env['XDG_CONFIG_HOME'] === undefined) {
+      // This env var isn't set on GitHub-hosted runners
+      this.onePasswordEnv.XDG_CONFIG_HOME = `${process.env['HOME']}/.config`
+    }
+  }
+
+  async setupAndInstallIfNeeded(): Promise<void> {
+    // Check if op is installed and download if necessary
+    const cachedOpDirectory = tc.find('op', ONE_PASSWORD_VERSION)
+    // This seems like a weird API, why not return undefined?
+    if (cachedOpDirectory !== '') {
+      core.addPath(cachedOpDirectory)
+    } else {
+      await install(ONE_PASSWORD_VERSION)
+    }
+  }
+
+  async signIn(
+    signInAddress: string,
+    emailAddress: string,
+    secretKey: string,
+    masterPassword: string
+  ): Promise<void> {
+    const env = this.onePasswordEnv
+    try {
+      const output = await execWithOutput(
+        'op',
+        [
+          'signin',
+          signInAddress,
+          emailAddress,
+          secretKey,
+          '--raw',
+          '--shorthand',
+          'github_action'
+        ],
+        {
+          env,
+          input: Buffer.alloc(masterPassword.length, masterPassword)
+        }
+      )
+      core.info('Successfully logged into 1Password')
+      const session = output.toString().trim()
+      core.setSecret(session)
+
+      this.onePasswordEnv.OP_SESSION_github_action = session
+    } catch (error) {
+      throw new Error(error)
+    }
+  }
+
+  async requestItems(itemRequests: parsing.ItemRequest[]): Promise<void> {
+    for (const itemRequest of itemRequests) {
+      try {
+        const itemsJSON = await this.getVault(itemRequest.vault)
+
+        const items: Item[] = JSON.parse(itemsJSON)
+        const uuid = items
+          .filter(item => item.overview.title === itemRequest.name)
+          .map(item => item.uuid)[0]
+
+        if (!uuid) {
+          throw new Error(
+            `Could not find item in vault \u001b[38;2;255;255;0m'${itemRequest.vault}' \u001b[38;2;255;255;255mwith name \u001b[38;2;255;255;0m'${itemRequest.name}'`
```

Comment:
For this and the other instances of shell escape codes, could you add a comment explaining what each does, or maybe we can use something like https://github.com/chalk/chalk to make it easier to work with?

---

When testing locally the color scheme may not be white-on-black like it is in GitHub's logs (and even that may change) so I wonder if this could be something more universally-legible like bold, instead of yellow? At least it was yellow-on-white for me when I ran it in act.

Reply:
Think `inverse` looks not bad. 

![image](https://user-images.githubusercontent.com/1119565/101865292-7a8b7080-3b3b-11eb-91e7-ed11648350d1.png)
![image](https://user-images.githubusercontent.com/1119565/101865303-8414d880-3b3b-11eb-98cf-eaad26ff9fa0.png)

Reply:
Added specific inverse like above for callouts. **Bold** for others 

![image](https://user-images.githubusercontent.com/1119565/101865732-ba068c80-3b3c-11eb-97e0-f713d2432ef8.png)

Reply:
Well that didn't work on github. :(

Reply:
Switched to `ansi-styles` which works on github

![image](https://user-images.githubusercontent.com/1119565/101919372-18a92600-3b90-11eb-9cb9-8994ae274b29.png)

Reply:
Oh weird that chalk didn't work, this looks good though! 👍 This would be a little harder to read as a dev but it might be better to remove the extra spaces around where the ANSI codes are interpolated, because those end up as duplicate spaces in the logs.